### PR TITLE
Random evaluation.py cleanup

### PR DESCRIPTION
**Pull request by mschwager**
Comments inline

### CODE REVIEW

File: semgrep/semgrep/evaluation.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def _evaluate_single_expression(
         )
         return output_ranges
     elif expression.operator == OPERATORS.WHERE_PYTHON:
-        if not flags or flags[RCE_RULE_FLAG] != True:
+        if not flags or not flags[RCE_RULE_FLAG]:
```
Comment (mschwager) [inline semgrep/semgrep/evaluation.py:95]: `flake8` suggestion

File: semgrep/semgrep/evaluation.py [hunk 2]
```diff
@@ -137,39 +137,35 @@ def _evaluate_single_expression(
         raise UnknownOperatorError(f"unknown operator {expression.operator}")
 
 
-# Given a `where-python` expression as a string and currently matched metavars,
```
Comment (mschwager) [inline semgrep/semgrep/evaluation.py:140]: This is just re-stating the `mypy` types and doesn't really contribute anything

File: semgrep/semgrep/evaluation.py [hunk 3]
```diff
@@ -137,39 +137,35 @@ def _evaluate_single_expression(
...
-# return whether the expression matches as a boolean
 def _where_python_statement_matches(
     where_expression: str, metavars: Dict[str, Any]
 ) -> bool:
     # TODO: filter out obvious dangerous things here
-    output_var = None
+    result = False
 
-    # HACK: we're executing arbitrary Python in the where-python,
```
Comment (mschwager) [inline semgrep/semgrep/evaluation.py:148]: "Be careful" doesn't really tell me anything here - the `# nosem` below tells me this is "approved insecure behavior"

File: semgrep/semgrep/evaluation.py [hunk 4]
```diff
@@ -137,39 +137,35 @@ def _evaluate_single_expression(
...
         new_last_line = f"{RETURN_VAR} = {lines[-1]}"
         lines[-1] = new_last_line
         to_eval = "\n".join(lines)
-        scope = {"vars": vars}
+        scope = {"vars": local_vars}
         # fmt: off
         exec(to_eval, scope)  # nosem: contrib.dlint.dlint-equivalent.insecure-exec-use, python.lang.security.audit.exec-detected.exec-detected
         # fmt: on
-        output_var = scope[RETURN_VAR]
+        result = scope[RETURN_VAR]  # type: ignore
```
Comment (mschwager) [inline semgrep/semgrep/evaluation.py:158]: It's impossible to statically determine the type here, the `isinstance` call below confirms the type for us :+1:

File: semgrep/semgrep/evaluation.py [hunk 5]
```diff
@@ -189,10 +185,6 @@ def safe_relative_to(a: Path, b: Path) -> Path:
         return a
 
 
-def should_exclude_this_path(path: Path) -> bool:
```
Comment (mschwager) [inline semgrep/semgrep/evaluation.py:192]: This function is no longer used after we removed the `--exclude-tests` flag.

File: semgrep/semgrep/evaluation.py [hunk 6]
```diff
@@ -137,39 +137,35 @@ def _evaluate_single_expression(
...
     where_expression: str, metavars: Dict[str, Any]
 ) -> bool:
     # TODO: filter out obvious dangerous things here
-    output_var = None
+    result = False
 
-    # HACK: we're executing arbitrary Python in the where-python,
-    # be careful my friend
-    vars = {k: v["abstract_content"] for k, v in metavars.items()}
+    local_vars = {k: v["abstract_content"] for k, v in metavars.items()}
```
Comment (mschwager) [inline semgrep/semgrep/evaluation.py:146]: [`vars`](https://docs.python.org/3/library/functions.html#vars) is a builtin Python function, let's avoid using it as a local variable name.

**Review by brendongo — APPROVED** [review]

**Pull request merged**